systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterIT.java

```diff
@@ -178,9 +178,9 @@ public void testZookeeperScaleUpScaleDown() {
         kubeClient.waitForPod(newZkPodName[1]);
 
         // check the new node is either in leader or follower state
-        waitForZkMntr(firstZkPodName, Pattern.compile("zk_server_state\\s+(leader|follower)"));
-        waitForZkMntr(newZkPodName[0], Pattern.compile("zk_server_state\\s+(leader|follower)"));
-        waitForZkMntr(newZkPodName[1], Pattern.compile("zk_server_state\\s+(leader|follower)"));
+        waitForZkMntr(firstZkPodName, "21810", Pattern.compile("zk_server_state\\s+(leader|follower)"));
+        waitForZkMntr(newZkPodName[0], "21811", Pattern.compile("zk_server_state\\s+(leader|follower)"));
+        waitForZkMntr(newZkPodName[1], "21812", Pattern.compile("zk_server_state\\s+(leader|follower)"));
```

Comment:
Just my thought about this `waitForZkMntr`. This method is used for waiting that a Zookeeper pod is running so it means that we know that it's about Zookeeper. Instead of passing the pod name we can pass the pod index (i.e. 0, 1 ...) and internally we can process the related name. We can use the pod index to compute the port as well, not having that as a new parameter. I did something similar with methods `listTopicsUsingPodCLI`, `createTopicUsingPodCLI` and so on. Just a thought ... no need to change the code if you are ok with this way.

Reply:
It's definitely a good idea. I did not notice that this code is in `AbstractClusterIT` and we have access to the cluster name, so we can generate zookeeper name, port, etc